### PR TITLE
Cgm -> TaC

### PR DESCRIPTION
**Pull request by ClumsyAlien**
After creating our fork of Mr.Crayfish's Gun mod we now need to port all of our current changes and updates over, one small step at a time!

### CODE REVIEW

File: src/main/java/com/tac/guns/entity/ProjectileEntity.java [hunk 1]
```diff
@@ -37,7 +36,6 @@
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.Pose;
-import net.minecraft.entity.item.ArmorStandEntity;
```
Comment (aemogie) [inline src/main/java/com/tac/guns/entity/ProjectileEntity.java]: is this cz of IntelliJ's "auto-optimize imports"?

File: src/main/java/com/tac/guns/client/render/gun/model/ak47_animation.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+            //Always push
+            matrices.pushPose();
+
+            //We're getting the cooldown tracker for the item - items like the sword, ender pearl, and chorus fruit all have this too.
+            CooldownTracker tracker = Minecraft.getInstance().player.getCooldowns();
+            float cooldownOg = tracker.getCooldownPercent(stack.getItem(), Minecraft.getInstance().getFrameTime());
+
+            // Math provided by Bomb787 on GitHub and Curseforge!!!
+            matrices.translate(0, 0, 0.190f * (-4.5 * Math.pow(cooldownOg-0.5, 2) + 1));
```
Comment (aemogie) [inline src/main/java/com/tac/guns/client/render/gun/model/ak47_animation.java:87]: @Bomb787 can u explain what exactly is going here XD
I probably can reduce code duplication but don't want to mess with stuff until I understand what's going on.

File: src/main/java/com/tac/guns/client/render/gun/model/m1851_animation.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        matrices.rotate(Vector3f.XP.rotationDegrees(90F));
+        RenderUtil.renderModel(SpecialModels.M1851_HAMMER.getModel(), stack, matrices, renderBuffer, light, overlay);
+
+        */
+
+        //Always pop
+        matrices.popPose();
+    }
+    private double easeInOutBack(double x) {
```
Comment (aemogie) [inline src/main/java/com/tac/guns/client/render/gun/model/m1851_animation.java:72]: make this static maybe? it doesn't use any fields so there's no real reason for it to be non-static.

File: src/main/java/com/tac/guns/entity/ProjectileEntity.java (hunk 1 above)
Comment (ClumsyAlien) [reply src/main/java/com/tac/guns/entity/ProjectileEntity.java]: I would think, if this is projectile entity this must be from where I started to test more complex bullet physics (Will be creating a separate branch for shortly), this is still an auto import and will be cleared before final push.

File: src/main/java/com/tac/guns/client/render/gun/model/ak47_animation.java (hunk 2 above)
Comment (ClumsyAlien) [reply src/main/java/com/tac/guns/client/render/gun/model/ak47_animation.java:87]: Sooooooooooooo ```cooldownOg``` is really just the percentage through the original cooldown of an item, and all we do is just splice the percentage into the equation, ``` 0.190f ``` is our multiplier for the equation.

File: src/main/java/com/tac/guns/client/render/gun/model/m1851_animation.java (hunk 3 above)
Comment (ClumsyAlien) [reply src/main/java/com/tac/guns/client/render/gun/model/m1851_animation.java:72]: Good point, 
**Idea** possibly create a ```FireAnimationHelper``` class to handle these sort of methods. 
In short yes ```easeInOutBack``` can be made static.

**Comment by ClumsyAlien** [comment]
An absolutely beautiful development period, 1.16.X is and will now stay as our Main home, 1.16.X public versions of the mod will be found here, all work must be done and only merged into Main, never work directly into Main.

**Pull request merged**